### PR TITLE
Unified bridge executors

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: ethToElrond/v2/bridge/bridgeExecutor.go [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+	return executor, nil
+}
+
+// CreateElrondToEthBridgeExecutor will create an Elrond->Eth bridge executor
+func CreateElrondToEthBridgeExecutor(args ArgsElrondToEthBridgeExectutor) (*bridgeExecutor, error) {
+	err := checkBaseArgs(args.ArgsBaseBridgeExecutor)
+	if err != nil {
+		return nil, err
+	}
+	if check.IfNil(args.TopologyProviderOnElrond) {
```
Comment (dragos-rebegea) [inline ethToElrond/v2/bridge/bridgeExecutor.go:71]: make also these 3 ifs a check function

File: ethToElrond/v2/bridge/bridgeExecutor.go [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+			return fmt.Errorf("%w for deposit %s, expected: %d", v2.ErrInvalidDepositNonce, dt.String(), startNonce)
+		}
+
+		startNonce++
+	}
+
+	return nil
+}
+
+// GetAndStoreActionIDForProposeTransferOnElrond fetches the action ID for ProposeTransfer by using the stored batch. Returns and stores the action ID
```
Comment (dragos-rebegea) [inline ethToElrond/v2/bridge/bridgeExecutor.go:171]: from a logical point of view, this stores first and then returns the action ID :smile:

File: ethToElrond/v2/bridge/bridgeExecutor.go [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+	if err != nil {
+		return v2.InvalidActionID, err
+	}
+
+	executor.actionID = actionID
+
+	return actionID, nil
+}
+
+// GetAndStoreActionIDForProposeSetStatusFromElrond fetches the action ID for SetStatus by using the stored batch. Returns and stores the action ID
```
Comment (dragos-rebegea) [inline ethToElrond/v2/bridge/bridgeExecutor.go:187]: same as above

File: ethToElrond/v2/bridge/bridgeExecutor.go [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+	defer timer.Stop()
+
+	select {
+	case <-ctx.Done():
+		executor.log.Debug("closing due to context expiration")
+	case <-timer.C:
+	}
+}
+
+// GetBatchStatusesFromEthereum gets statuses for the batch id
```
Comment (dragos-rebegea) [inline ethToElrond/v2/bridge/bridgeExecutor.go:294]:
```suggestion
// GetBatchStatusesFromEthereum gets statuses for the batch
```

File: ethToElrond/v2/bridge/bridgeExecutor.go [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+
+	statuses, err := executor.ethereumClient.GetTransactionsStatuses(ctx, executor.batch.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return statuses, nil
+}
+
+// WasActionPerformedOnElrond will return true if the action was already performed
```
Comment (dragos-rebegea) [inline ethToElrond/v2/bridge/bridgeExecutor.go:308]: make all comments to have the same verbal tense

File: ethToElrond/v2/signaturesHolder_test.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+	return &core.EthereumSignature{
+		Signature:   []byte(fmt.Sprintf("sig %d", index)),
+		MessageHash: []byte("message hash"),
+	}
+}
+
+func TestSignatureHolder_ProcessNewMessage(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil messages", func(t *testing.T) {
```
Comment (dragos-rebegea) [inline ethToElrond/v2/signaturesHolder_test.go:33]: add t.Parallel()

File: testsCommon/bridgeV2/ethereumClientStub.go [hunk 7]
```diff
@@ -15,6 +16,36 @@ type EthereumClientStub struct {
 	BroadcastSignatureForMessageHashCalled     func(msgHash common.Hash)
 	ExecuteTransferCalled                      func(ctx context.Context, msgHash common.Hash, batch *clients.TransferBatch, quorum int) (string, error)
 	GetMaxNumberOfRetriesOnQuorumReachedCalled func() uint64
+	GetTransactionsStatusesCalled              func(ctx context.Context, batchId uint64) ([]byte, error)
+	GetQuorumSizeCalled                        func(ctx context.Context) (*big.Int, error)
+	IsQuorumReachedCalled                      func(ctx context.Context, msgHash common.Hash) (bool, error)
+}
+
+// GetTransactionsStatuses -
```
Comment (dragos-rebegea) [inline testsCommon/bridgeV2/ethereumClientStub.go:24]: put the stub methods in order

File: factory/ethElrondBridgeComponents.go [hunk 8]
```diff
@@ -310,7 +310,7 @@ func (components *ethElrondBridgeComponents) createEthereumClient(args ArgsEther
 		Broadcaster:               components.broadcaster,
 		PrivateKey:                privateKey,
 		TokensMapper:              tokensMapper,
-		SignatureHolder:           &disabledSignatureHolder{}, // TODO replace this with the real component
```
Comment (iulianpascalau) [inline factory/ethElrondBridgeComponents.go:313]: this instance should be passed to the broadcaster instance as the broadcaster handles the new messages flow.

File: ethToElrond/v2/bridge/bridgeExecutor.go [hunk 9]
```diff
@@ -95,14 +55,22 @@ func checkBaseArgs(args ArgsBaseBridgeExecutor) error {
 	if check.IfNil(args.EthereumClient) {
 		return v2.ErrNilEthereumClient
 	}
+	if check.IfNil(args.TopologyProvider) {
+		return v2.ErrNilTopologyProvider
+	}
+	if args.TimeForTransferExecution.Seconds() == 0 {
```
Comment (iulianpascalau) [inline ethToElrond/v2/bridge/bridgeExecutor.go:61]: or could have been a comparison with a defined const like 1 second

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**